### PR TITLE
implement syscall 605

### LINKED ISSUE #415: Unimplemented Syscall 605
```
++++++++++++++++++ I [00:00:00:01:611]:0x0000000000003c08:kernel\src\syscalls\mod.rs:51
Syscall called: 605
++++++++++++++++++ P [00:00:00:01:611]:0x0000000000003c08:kernel\src\syscalls\mod.rs:55
not yet implemented: syscall 605 at 0x1e5a on /system/common/lib/libkernel.sprx
```

Reached after bypassing 487 and 596.

CUSTOM PS4 SYSCALL

Syscall ID | PS4 FW Version | Syscall name | Syscall Prototype
-- | -- | -- | --
605| <=1.01? | sys_workaround8849 | -

<details>
  <summary>
    Function Code:
  </summary>
  <code>__int64 __usercall sys_workaround8849@<rax>(__int64 a1@<rdi>, unsigned int *a2@<rsi>, __int64 a3@<r12>, _QWORD *a4@<r13>, __int64 a5@<r15>, double a6@<xmm0>)
{
  __int64 v6; // rbx
  unsigned int v7; // edi
  __int64 result; // rax
  int v9; // [rsp+Ch] [rbp-14h]
  v6 = a1;
  v7 = *a2;
  result = 22LL;
  if ( *a2 <= 0x780263FF )
  {
    if ( v7 != 427884544 )
      return result;
LABEL_7:
    v9 = 0;
    result = sceRegMgrGetInt(v7, (char *)&v9, a3, a4, a5, a6);
    *(_QWORD *)(v6 + 896) = v9;
    return result;
  }
  if ( v7 == 2013422592 || v7 == 2013430528 || v7 == 2013432320 )
    goto LABEL_7;
  return result;
}
</code>
</details>

### PR DESCRIPTION
**Pull request by SuchAFuriousDeath**
Should fix #415 once syscall 596 is implemented.
I'm not sure about the unwrap after get_int.

### CODE REVIEW

**Comment by ultimaweapon** [comment]
Here is a list of registry keys: https://psdevwiki.com/ps4/Talk:Registry

**Comment by SuchAFuriousDeath** [comment]
Added them. Hope I didn't mess up the decimal to hex conversion.

File: src/kernel/src/regmgr/mod.rs [hunk 1]
```diff
@@ -299,6 +310,28 @@ impl RegMgr {
...
+
+        match key {
+            RegKey::NP_DEBUG
+            | RegKey::BROWSER_DEBUG_NOTIFICATION
+            | RegKey::DEVENV_TOOL_TRC_NOTIFY
+            | RegKey::DEVENT_TOOL_USE_DEFAULT_LIB
+            | RegKey::DEVENV_TOOL_SYS_PRX_PRELOAD => {
+                let val = self.get_int(key).unwrap();
+
+                Ok(val.into())
```
Comment (ultimaweapon) [inline src/kernel/src/regmgr/mod.rs:330]: `td_retval` will be ignored if non-zero is returned from the syscall handler. See https://github.com/freebsd/freebsd-src/blob/release/9.1.0/sys/amd64/amd64/vm_machdep.c#L380 for how it is handled.

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

**Comment by SuchAFuriousDeath** [comment]
Just FYI, I'm aware that I'm missing a param, check in the get_int function. Just so you don't have to point that out.

**Comment by SuchAFuriousDeath** [comment]
Also, which issue will you focus on next?

File: src/kernel/src/regmgr/mod.rs (hunk 1 above)
Comment (SuchAFuriousDeath) [reply src/kernel/src/regmgr/mod.rs:330]: Thanks, I'll look into it when I have some time.

**Comment by ultimaweapon** [comment]
Not planned yet. Now I have a full-time job so I only have weekend on work on this.

**Comment by SuchAFuriousDeath** [comment]
#MeToo

**Comment by SuchAFuriousDeath** [comment]
Just pushed a new commit, take a look when you have time. Honestly, I'm a bit confused by all the return values and conditions

**Comment by ultimaweapon** [comment]
Which ones are you confusing?

**Comment by SuchAFuriousDeath** [comment]
What does the Ok variant of the return value of set_int represent?

**Comment by ultimaweapon** [comment]
It represent the return value of `regMgrComSetReg` if it is `>= 0`.

**Review by ultimaweapon — APPROVED** [review]

**Pull request merged**